### PR TITLE
Client library, OpenAPI Spec and Automated Testing

### PR DESCRIPTION
**Pull request by rcgoodfellow**
- OpenAPI spec generation.
- A progenitor-based rust client for admin operations.
- Simple integration tests that cover the following in the server and the generated client.
  - CRUD operations for AAAA and SRV records
  - Resolve requests for AAAA and SRV records
- A Github Workflow CI spec that covers formatting, clippy, building, and testing.

### CODE REVIEW

File: openapi/toy-dns.json [hunk 1]
```diff
@@ -0,0 +1,166 @@
+{
```
Comment (davepacheco) [inline openapi/toy-dns.json:1]: What's the difference between this toy-dns.json and the one at the top level?

File: Cargo.toml [hunk 2]
```diff
@@ -6,14 +6,23 @@ edition = "2021"
...
 toml = "0.5"
 trust-dns-server = "0.21"
 trust-dns-proto = "0.21"
+trust-dns-resolver = "0.21"
 pretty-hex = "0.2.1"
 schemars = "0.8"
 sled = "0.34"
-serde_json = "1"
+serde_json = "1.0"
+reqwest = { version = "0.11", features = ["json", "rustls-tls", "stream"] }
```
Comment (davepacheco) [inline Cargo.toml:26]: Should any of these be dev-dependencies instead?

**Review by davepacheco — COMMENTED** [review]
Nice!  Happy to land this whenever but I just had two questions below.

File: openapi/toy-dns.json (hunk 1 above)
Comment (rcgoodfellow) [reply openapi/toy-dns.json:1]: The difference is me forgetting to `mv toy-dns.json openapi/` after generating an update version :/

Comment (rcgoodfellow) [reply openapi/toy-dns.json:1]: Fixed in d8a9c22

File: Cargo.toml (hunk 2 above)
Comment (rcgoodfellow) [reply Cargo.toml:26]: Yup, totally. I found a few that are only used in tests.

Comment (rcgoodfellow) [reply Cargo.toml:26]: Updated in 3c4acfa

**Pull request merged**